src/Agent.Plugins/TfsVCCliManager.cs

```diff
@@ -131,7 +131,7 @@ protected async Task RunCommandAsync(FormatFlags formatFlags, bool quiet, int re
 
                     int sleep = Math.Min(200 * (int)Math.Pow(5, attempt), 30000);
                     ExecutionContext.Output($"Sleeping for {sleep} ms");
-                    Thread.Sleep(sleep);
+                    await Task.Delay(sleep);
```

Comment:
Probably better?

Task.Sleep holds the thread so it's exact.  Task.Delay gives it up and will approx get it back in the sleep time.  However if the process is really busy it may take a while to get the thread back.  Sounds like a maxed off process isn't a concern here so Task.Delay makes sense.